library/model/borrowing.py

```diff
@@ -0,0 +1,54 @@
+from library.model.book import Book
+from library.model.user import User
+from datetime import datetime, timedelta
+from library.persistence.storage import LibraryRepository
+
+
+class Borrowing:
+    book: Book
+    user: User
+    due_date: datetime
+    fee: float
+
+    def __init__(self, book:Book, user:User, due_date: datetime):
+        self.book = book
+        self.user = user
+        self.fee = book.get_weekly_fee()
+        self.due_date = due_date
+
+    def open_borrowing(self):
+        try:
+            if self.book.can_borrow():
+                if self.book._book_type == "Paper":
+                    self.book.borrowed_items += 1
+                self._update_library_repository()
+            return
+        except AttributeError:
+            raise AttributeError("Attribute error")
+        except ValueError:
+            raise ValueError("Book cannot be borrowed")
+
+    def renew_borrowing(self):
+        self.due_date += timedelta(days=7)
+        self.fee += self.book.get_weekly_fee()
+
+    def close_borrowing(self):
+        if self.book._book_type == "Paper":
+            self.book.borrowed_items -= 1
+        self._update_library_repository()
```

Comment:
Ideally, a class should not update itself in a database. It makes it really hard to test because the side-effects of the method call are unknown